src/compiler/watch.ts

```diff
@@ -322,21 +322,21 @@ namespace ts {
 
             if (hasChangedCompilerOptions) {
                 newLine = getNewLineCharacter(compilerOptions, system);
+                if (changesAffectModuleResolution(program && program.getCompilerOptions(), compilerOptions)) {

```

Comment:
Nit: Would prefer `program && changesAffectModuleResolution(program.getCompilerOptions(), compilerOptions)` since if `!program`, `changesAffectModuleResolution` has nothing it can do.

Reply:
+1